src/query/plan/operator.cpp

```diff
@@ -1694,6 +1718,37 @@ void CheckWeightType(TypedValue current_weight, utils::MemoryResource *memory) {
   }
 }
 
+void ValidateWeightTypes(const TypedValue &lhs, const TypedValue &rhs) {
+  if ((lhs.IsNumeric() && rhs.IsNumeric()) || (lhs.IsDuration() && rhs.IsDuration()) ||
+      // We make implicit conversion of 0 as int to Duration to work correctly in weight lambda at the path begin.
+      (lhs.IsInt() && lhs.ValueInt() == 0 && rhs.IsDuration()) ||
```

Comment:
Are we sure only `Int` should be supported?

Reply:
Updated the PR. Now we don't have this block of code because we don't need it.